server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java

```diff
@@ -657,6 +658,12 @@ private static ServerAddress startTServer(ThriftServerType serverType, TimedProc
       }
     }).start();
 
+    while (!finalServer.isServing()) {
+      // Wait for the thread to start and for the TServer to start
+      // serving events
+      UtilWaitThread.sleep(10);
+    }
```

Comment:
Could add a defensive check for future code changes.

```suggestion
       Preconditions.checkState(!finalServer.getShouldStop());
    }
```